lab_3/main.py

```diff
@@ -100,33 +137,33 @@ def encode(storage_instance, corpus) -> list:
             corpus[i][k] = storage_instance[corpus[i][k]]
     return corpus
 
-def split_by_sentence(text: str) -> list:
-    if text == '' or text == None:
+def split_by_sentence(REFERENCE_TEXT: str) -> list:
+    if REFERENCE_TEXT == '' or REFERENCE_TEXT == None:
         return []
     else:
-        if "\n" in text:
-            text = text.replace("\n", " ")
-        while "  " in text:
-            text = text.replace("  ", " ")
-        text = list(text)
-        text = list(filter(lambda x: x not in undesired, text))
-        text.remove(text[-1])
-        for i in range(len(text)):
-            if text[i] in punct_marks:
-                text[i] = '.'
-        text = ''.join(text)
-        text = text.split(".")
-        for i in range(0, len(text)):
-            if text[i].isalpha:
-                text[i] = text[i].lower()
-        if len(text) == 1:
+        if "\n" in REFERENCE_TEXT:
+            REFERENCE_TEXT = REFERENCE_TEXT.replace("\n", " ")
+        while "  " in REFERENCE_TEXT:
+            REFERENCE_TEXT = REFERENCE_TEXT.replace("  ", " ")
+        REFERENCE_TEXT = list(REFERENCE_TEXT)
+        REFERENCE_TEXT = list(filter(lambda x: x not in undesired, REFERENCE_TEXT))
+        REFERENCE_TEXT.remove(REFERENCE_TEXT[-1])
+        for i in range(len(REFERENCE_TEXT)):
+            if REFERENCE_TEXT[i] in punct_marks:
+                REFERENCE_TEXT[i] = '.'
+        REFERENCE_TEXT  = ''.join(REFERENCE_TEXT)
+        REFERENCE_TEXT = REFERENCE_TEXT.split(".")
+        for i in range(0, len(REFERENCE_TEXT)):
+            if REFERENCE_TEXT[i].isalpha:
+               REFERENCE_TEXT[i] = REFERENCE_TEXT[i].lower()
+        if len(REFERENCE_TEXT) == 1:
             return []
-        for i in range(0, len(text)):
-            text[i] = text[i].split(" ")
-            for k in range(0, len(text[i]) - 1):
-                if text[i][k] == '':
-                    text[i].remove(text[i][k])
-                text[i][k].lower()
-            text[i].insert(0, "<s>")
-            text[i].insert(len(text[i]), "</s>")
-        return text
\ No newline at end of file
+        for i in range(0, len(REFERENCE_TEXT)):
+            REFERENCE_TEXT[i] = REFERENCE_TEXT[i].split(" ")
+            for k in range(0, len(REFERENCE_TEXT[i]) - 1):
+                if REFERENCE_TEXT[i][k] == '':
+                   REFERENCE_TEXT[i].remove(REFERENCE_TEXT[i][k])
```

Comment:
@kazadvornov you can not remove elements from text you are iterating over - you will get IndexError: list index out of range